cli/mender-artifact/artifacts.go

```diff
@@ -132,61 +123,40 @@ func getKey(keyPath string) ([]byte, error) {
 	return key, nil
 }
 
-func unpackArtifact(name string) (string, error) {
+func unpackArtifact(name string) (ua *unpackedArtifact, err error) {
+	ua = &unpackedArtifact{
+		origPath: name,
+	}
+
 	f, err := os.Open(name)
 	if err != nil {
-		return "", errors.Wrapf(err, "Can not open: %s", name)
+		return nil, errors.Wrapf(err, "Can not open: %s", name)
 	}
 	defer f.Close()
 
 	// initialize raw reader and writer
```

Comment:
Not your work, I know, but I don't really see a writer initialized here?

Reply:
Indeed. I removed the whole comment, it was stating the obvious! :-)